src/ImageCarouselReact/widget/components/Carousel.tsx

```diff
@@ -0,0 +1,538 @@
+     * First argument is the target index i.e Index of the item that triggered it
+     */
+    onSlide?: Function;
+    onSlideEnd?: Function;
+    /**
+     * Index of the current image being showed on the carousel
+     *
+     * @type {number}
+     * @memberOf ICarouselProps
+     */
+    activeIndex?: number;
+    /**
+     * Index of the image that should be shown first
+     *
+     * @type {number}
+     * @memberOf ICarouselProps
+     */
+    defaultActiveIndex?: number;
+    /**
+     * Direction in which the carousel should auto-transition
+     *
+     * @type {Direction}
+     * @memberOf ICarouselProps
+     */
+    direction?: Direction;
+    /**
+     * Icon for the "prev" navigation control
+     *
+     * @type {JSX.Element}
+     * @memberOf ICarouselProps
+     */
+    prevIcon?: JSX.Element;
+    /**
+     * Icon for the "next" navigation control
+     *
+     * @type {JSX.Element}
+     * @memberOf ICarouselProps
+     */
+    nextIcon?: JSX.Element;
+    className?: string;
+    /**
+     * Contains Base properties eg. Base CSS class and prefix for the component
+     * Use only when you want to specify a non-bootstrap properties
+     *
+     * @type {string}
+     * @memberOf ICarouselProps
+     */
+    bsProps?: IBootstrapProps;
+    elementProps?: {};
+};
+
+export type Direction = "prev" | "next";
+
+export interface ICarouselState {
+    activeIndex?: number;
+    previousActiveIndex?: number;
+    direction?: Direction;
+}
+
+export interface ICarouselEvent<T> extends React.MouseEvent<T> {
+    direction: Direction;
+}
+
+class Carousel extends React.Component<ICarouselProps, ICarouselState> {
+    public static defaultProps: ICarouselProps = {
+        bsProps: {
+            bsClass: "carousel",
+        },
+        interval: 5000,
+        nextIcon: <Glyphicon glyph="chevron-right" />,
+        pauseOnHover: true,
+        prevIcon: <Glyphicon glyph="chevron-left" />,
+        showControls: true,
+        showIndicators: true,
+        slide: true,
+        wrap: true,
+    };
+    private isUnmounted: boolean;
+    private timeout: number;
+    private isPaused: boolean;
+    private loggerNode: string;
+    constructor(props: ICarouselProps, context: Carousel) {
+        super(props, context);
+        this.loggerNode = "Carousel";
+        logger.debug(this.loggerNode + " .constructor");
+        // bind context
+        this.handleMouseOver = this.handleMouseOver.bind(this);
+        this.handleMouseOut = this.handleMouseOut.bind(this);
+        this.handlePrev = this.handlePrev.bind(this);
+        this.handleNext = this.handleNext.bind(this);
+        this.handleItemAnimateOutEnd = this.handleItemAnimateOutEnd.bind(this);
+
+        const { defaultActiveIndex } = props;
+
+        this.state = {
+            activeIndex: defaultActiveIndex != null ? defaultActiveIndex : 0,
+            direction: null,
+            previousActiveIndex: null,
+        };
+
+        this.isUnmounted = false;
```

Comment:
Not true, there is an instance now but it's not necessarily mounted. That's only done when `componentDidMount` is called.